firedrake/projection.py

```diff
@@ -174,19 +181,19 @@ def A(self):
         if not mixed and isinstance(F.finat_element, HDivTrace):
             if F.extruded:
                 a = (
-                    firedrake.inner(u, v)*firedrake.ds_t(degree=self.quadrature_degree)
-                    + firedrake.inner(u, v)*firedrake.ds_v(degree=self.quadrature_degree)
-                    + firedrake.inner(u, v)*firedrake.ds_b(degree=self.quadrature_degree)
-                    + firedrake.inner(u('+'), v('+'))*firedrake.dS_h(degree=self.quadrature_degree)
-                    + firedrake.inner(u('+'), v('+'))*firedrake.dS_v(degree=self.quadrature_degree)
+                    firedrake.inner(u, v)*firedrake.ds_t
+                    + firedrake.inner(u, v)*firedrake.ds_v
+                    + firedrake.inner(u, v)*firedrake.ds_b
+                    + firedrake.inner(u('+'), v('+'))*firedrake.dS_h
+                    + firedrake.inner(u('+'), v('+'))*firedrake.dS_v
                 )
             else:
                 a = (
-                    firedrake.inner(u, v)*firedrake.ds(degree=self.quadrature_degree)
-                    + firedrake.inner(u('+'), v('+'))*firedrake.dS(degree=self.quadrature_degree)
+                    firedrake.inner(u, v)*firedrake.ds
+                    + firedrake.inner(u('+'), v('+'))*firedrake.dS
                 )
```

Comment:
I can't leave a suggestion because some lines are not modified, but (in my opinion) this would be cleaner by making it obvious that the if-statement is only on the measure, not the integrand.

```python
if F.extruded:
    ds = firedrake.ds_t + firedrake.ds_v + firedrake.ds_b
    dS = firedrake.dS_h + firedrake.dS_v
else:
    ds = firedrake.ds
    dS = firedrake.dS

a = (
    firedrake.inner(u, v)*ds
    + firedrake.inner(u('+'), v('+'))*dS
)
```